File: src/component_info.cpp
#include "component_info.h"


ComponentInfo::ComponentInfo() : componentInfo(gcs::ffi::create_component_info(0)){

}

void ComponentInfo::set_component_info(gcs::ffi::ComponentInfo* argComponentInfo) {
    componentInfo = rust::box<gcs::ffi::ComponentInfo>::from_raw(argComponentInfo);
}

gcs::ffi::ComponentInfo &ComponentInfo::getComponentInfo() {
    return componentInfo.operator*();
}

void ComponentInfo::_bind_methods() {
}

File: src/component/component_info.rs
use cxx::{type_id, ExternType};

#[derive(Hash, Eq, PartialEq, Debug, Clone, Copy)]
pub struct ComponentInfo {
    pub hash: u64,
}

unsafe impl ExternType for ComponentInfo {
    type Id = type_id!("gcs::ffi::ComponentInfo");
    type Kind = cxx::kind::Trivial;
}

pub(crate) fn create_component_info(hash: u64) -> Box<ComponentInfo> {
    Box::new(ComponentInfo { hash })
}
